modules/base/acl/module.py

```diff
@@ -25,494 +23,606 @@ def __init__(self, bot):
-
-        # add relationships to group objects
-        for group in groups:
-            group.children = relationships[group.name]
-            group.level = 0
-
-        def bfs(queue: List[ACL_group]) -> List[ACL_group]:
-            visited: List[ACL_group] = list()
-            while queue:
-                group = queue.pop(0)
-                if group not in visited:
-                    visited.append(group)
-                    # build levels for indentation
-                    for child in group.children:
-                        child.level = group.level + 1
-                    queue = group.children + queue
-            return visited
-
-        result = ""
-        template = "{group_id:<2} {name:<20} {role:<18}"
-        for group in bfs(groups):
-            result += "\n" + template.format(
-                group_id=group.idx,
-                name="  " * group.level + group.name,
-                role=group.role_id,
-            )
+        class Item:
+            def __init__(self, mapping: ACLevelMappping):
+                self.level = mapping.level.name
+                role = ctx.guild.get_role(mapping.role_id)
+                self.role = getattr(role, "name", str(mapping.role_id))
 
-        await ctx.reply(f"```{result}```")
+        mappings = ACLevelMappping.get_all(ctx.guild.id)
 
-    @commands.check(check.acl)
-    @acl_group.command(name="get")
-    async def acl_group_get(self, ctx, name: str):
-        """Get ACL group."""
-        group = ACL_group.get(ctx.guild.id, name)
-        if group is None:
-            await ctx.reply(_(ctx, "No such group exists."))
+        if not mappings:
+            await ctx.reply(_(ctx, "No mappings have been set."))
             return
 
-        await ctx.reply(embed=self.get_group_embed(ctx, group))
-
-    @commands.check(check.acl)
-    @acl_group.command(name="add")
-    async def acl_group_add(self, ctx, name: str, parent: str, role_id: int):
-        """Add ACL group.
+        mappings = sorted(mappings, key=lambda m: m.level.name)[::-1]
+        items = [Item(mapping) for mapping in mappings]
 
-        :param name: string matching ``[a-zA-Z-]+``.
-        :param parent: ACL parent group name.
-        :param role_id: Discord role ID.
+        table: List[str] = utils.text.create_table(
+            items,
+            header={
+                "role": _(ctx, "Role"),
+                "level": _(ctx, "Level"),
+            },
+        )
 
-        To unlink the group from the parent, set it to "".
+        for page in table:
+            await ctx.send("```" + page + "```")
 
-        To set up virtual group with no link to Discord roles, set ``role_id``
-        to ``0``.
-        """
-        RE_NAME = r"[a-zA-Z-]+"
-        if re.fullmatch(RE_NAME, name) is None:
+    @check.acl2(check.ACLevel.GUILD_OWNER)
+    @acl_mapping_.command(name="add")
+    async def acl_mapping_add(self, ctx, role: nextcord.Role, level: str):
+        """Add ACL level to role mappings."""
```

Comment:
Is it possible to give user more information about those levels?

Reply:
There will be documentation on this topic. But we may do something to make it easier.

Reply:
I think just simple list of levels or command to list levels would be good enough.